Linux AppImage Documentation (#245)
owncloud/docs-client-desktop: modules/ROOT/pages/installing.adoc

= Installing the Desktop Client
:toc: right
:toclevels: 4
:ms-remove-url: https://docs.microsoft.com/en-us/windows/win32/msi/remove
:ms-adddefault-url: https://docs.microsoft.com/en-us/windows/win32/msi/adddefault
:desktop-clients-url: https://owncloud.com/desktop-app/
:appimage-wikipedia-url: https://en.wikipedia.org/wiki/AppImage
:gnome-keyring-url: https://wiki.gnome.org/Projects/GnomeKeyring/
:kwalletmanager-url: https://utils.kde.org/projects/kwalletmanager/
:ubuntu-lts-url: https://wiki.ubuntu.com/LTS

> :libfuse2-url: https://docs.appimage.org/user-guide/troubleshooting/fuse.html#setting-up-fuse-2-x-alongside-of-fuse-3-x-on-recent-ubuntu-22-04-debian-and-their-derivatives

Author: This is not an issue, because when an anchor gets invalid you automatically get on the top of the page.

:install-appimage-url: https://docs.appimage.org/introduction/quickstart.html#ref-quickstart

> :description: The Desktop Client enables users to access and sync files and folders form their ownCloud, work on remote files right from the desktop as if they were stored on their computer – because they are. Continuous synchronization to and from the ownCloud server provides ease of use combined with comprehensive access control.

Reviewer: ```suggestion
:description: The Desktop Client enables users to access and sync files and folders from their ownCloud, work on remote files right from the desktop as if they were stored on their computer – because they are. Continuous synchronization to and from the ownCloud server provides ease of use combined with comprehensive access control.
```

== Introduction

{description}

== System Requirements and Installation

> You can download the latest version of the ownCloud Desktop Synchronization Client from the {desktop-clients-url}[Desktop Client Download] page. There are clients for _Linux_, _macOS_, and _Microsoft Windows_ available. When using Linux, you can also use Appimage.

Reviewer: ```suggestion
You can download the latest version of the ownCloud Desktop Client from the {desktop-clients-url}[Desktop Client Download] page. There are clients for _Linux_, _macOS_, and _Microsoft Windows_ available. When using Linux, you can also use Appimage.
```

=== Installation on Linux

==== Full installation

> Linux users must follow the instructions on the {desktop-clients-url}[download] page to add the appropriate repository for their Linux distribution, install the signing key, and use their package managers to install the desktop sync client. Linux users will also update their sync clients via package manager, the client will display a notification when an update is available.

Reviewer 2: ```suggestion
Linux users must follow the instructions on the {desktop-clients-url}[download] page to add the appropriate repository for their Linux distribution, install the signing key and use their package managers to install the desktop sync client. Linux users will also update their sync clients via package manager. The client will display a notification when an update is available.
```

Linux users must also have a password manager enabled, such as {gnome-keyring-url}[GNOME Keyring] or {kwalletmanager-url}[KWallet], so that the sync client can login automatically.

You will also find links to source code archives and older versions on the download page.

==== AppImage

> Starting with Desktop version 2.9, an {appimage-wikipedia-url}[AppImage] build of the ownCloud Desktop Client is available to support more Linux platforms. You can download the AppImage right at the {desktop-clients-url}[Linux section of the Download Desktop App] page.

Reviewer: ```suggestion
Starting with Desktop Client version 2.9, an {appimage-wikipedia-url}[AppImage] build of the ownCloud Desktop Client is available to support more Linux platforms. You can download the AppImage right at the {desktop-clients-url}[Linux section of the Download Desktop Client] page.
```

> Starting with Desktop version 2.9, an {appimage-wikipedia-url}[AppImage] build of the ownCloud Desktop Sync Client is available to support more Linux platforms. You can download the AppImage right at the {desktop-clients-url}[Linux section of the Download Desktop App] page.

Reviewer: ```suggestion
Starting with Desktop version 2.9, an {appimage-wikipedia-url}[AppImage] build of the ownCloud Desktop Client is available to support more Linux platforms. You can download the AppImage right at the {desktop-clients-url}[Linux section of the Download Desktop App] page.
```

> AppImage is a different way to install Linux applications - not several files (packages) in several places but one file (ending .AppImage) containing the application and all necessary dependency and libraries. There is only a single application AppImage which runs on all (modern) Linux platforms.

Reviewer: ```suggestion
AppImage is a different way to install Linux applications - not several files (packages) in several places but one file (ending with .AppImage) containing the application and all necessary dependencies and libraries. There is only a single application AppImage which runs on all (modern) Linux platforms.
```

Our description focuses on the GNOME desktop.

Known limitations for the 2.10.x AppImage::

> * CentOS 7 is currently not supported

Reviewer 2: ```suggestion
* CentOS 7 is currently not supported.
```

> * For Ubuntu 22.04 or Debian 11, you need installing `libfuse2` as prerequisite. For details see

Reviewer: ```suggestion
* For Ubuntu 22.04 or Debian 11, you need to install `libfuse2` as prerequisite. For details see
```

> issue with `libfuse` on Ubuntu >=22.04 or Debian 11 {libfuse2-url}[Setting up FUSE 2.x alongside of FUSE 3.x on recent Ubuntu (>=22.04), Debian and their derivatives]

Reviewer 2: ```suggestion
issue with `libfuse` on Ubuntu >=22.04 or Debian 11 {libfuse2-url}[Setting up FUSE 2.x alongside of FUSE 3.x on recent Ubuntu (>=22.04), Debian and their derivatives].
```

* The file browser integration, which means overlay icons and the context menu is not included in the AppImage. You need to install file browser extension packages manually.

> * AppImages do not start automatically. You have to configure your desktop to automatically start the Desktop Client App when logging in. For GNOME, search for _startup applications_ in the desktop menu. 

Reviewer 3: Note for 3.0: the AppImage will set up the autostart correctly again. This limitation really just affects 2.10.

Author: As written above, this is the documentation for next which will be backported to 2.10 and 2.9. We will review when 3.0 is out. Standard doc process.

> * There is no automatic updating. Any update is like installing the AppImage.

Reviewer 4: Thats not longer the case for 3.0

Installing _libfuse2_ if required::
* Check if `libfuse2` is already installed:
+
[source,bash]
----
dpkg -l libfuse2
----
+
* Check if there is an installation candidate for `libfuse2`:
+
[source,bash]
----
sudo apt-cache show libfuse2
----
+

> * Install `libfuse2`

Reviewer 2: ```suggestion
* Install `libfuse2`:
```

+
[source,bash]
----
sudo apt install libfuse2
----

Install and run the Desktop App AppImage::

> The example below uses the terminal but you also can use the GUI, for details see {install-appimage-url}[How to run an AppImage].

Reviewer 2: ```suggestion
The example below uses the terminal but you can also use the GUI. For details see {install-appimage-url}[How to run an AppImage].
```

+
* Go to the {desktop-clients-url}[download] page and download the recent AppImage to your home directory into the Applications folder. Replace the URL from the example with the actual URL from the download page.
+
[source,bash]
----

> mkdir -p ~/Applications

Reviewer 4: Any reason for this 'Mac'ism

Author: This is standard linux writing

Reviewer 4: I mean the Applications dir

Reviewer 3: AppImageLauncher (and some other tools) use this directory by default.

=== Installation on Mac OS X and Windows

Installation on Mac OS X and Windows is the same as for any software application: download the installer,  double-click it to launch the installation and follow the installation wizard. After it is installed and configured the Desktop App will automatically keep itself updated; see `autoupdate` for more information.

=== Installation on Linux

> * Independent if installing the Desktop App natively or as AppImage, you can add command line parameters when starting up post installation. One parameter to mention is the option `-s`. This option forces to show the settings page on startup. While not necessary on general usage, it can be helpful if systray icons are not available in your desktop system.

Reviewer: ```suggestion
* Independent of installing the Desktop App natively or as an AppImage, you can add command line parameters when starting up post installation. One parameter to mention is the option `-s`. This option forces the settings page to be shown on startup. While not necessary on general usage, it can be helpful if systray icons are not available in your desktop system.
```

[source,bash]
----
sudo apt install gnome-shell-extension-appindicator
----

When the extension is installed, you need to restart the Gnome shell. To do so, hit btn:[Alt+F2] on the keyboard, type btn:[r] and  btn:[Enter]. Now you will see the new extensions added to the extensions list. To enable an extension, switch the button on the right to the btn:[On] position. For the particular systray icon extension, the name for all distributions is **Ubuntu appindicators** because Ubuntu has developed this extension. Enable it for showing systray icons.

image::installing/gnome-shell-extension-appindicator-selector.png[width=350%,pdfwidth=60%]
--

==== Full installation

Linux users must follow the instructions on the {desktop-clients-url}[download] page to add the appropriate repository for their Linux distribution, install the signing key and use their package managers to install the Desktop App. Linux users will also update their Desktop App via package manager. The Desktop App will display a notification when an update is available.

You will also find links to source code archives and older versions on the download page.

==== AppImage

> Starting with Desktop App version 2.9, an {appimage-wikipedia-url}[AppImage] build of the ownCloud Desktop App is available to support more Linux platforms. You can download the AppImage right at the {desktop-clients-url}[Linux section of the Download Desktop App] page.

Reviewer: ```suggestion
Starting with Desktop App version 2.9, an {appimage-wikipedia-url}[AppImage] build of the ownCloud Desktop App is available to support more Linux platforms. You can download the AppImage at the {desktop-clients-url}[Linux section of the Download Desktop App] page.
```

> AppImage is an alternative way to use Linux applications -- instead of having multiple files respectively packages in several places, the entire application is contained in a single file ending with an `.AppImage` suffix, including all necessary dependencies and libraries. ownCloud provides a single AppImage which runs on all (modern) Linux platforms.

Reviewer: ```suggestion
AppImage is an alternative way to use Linux applications -- instead of having multiple files in several places making up a package, the entire application is contained in a single file ending with an `.AppImage` suffix, including all necessary dependencies and libraries. ownCloud provides a single AppImage which runs on all (modern) Linux platforms.
```

Known limitations for the 2.10.x AppImage::
* CentOS 7 is currently not supported.

* For Ubuntu 22.04 or Debian 11, you need to install `libfuse2` as a prerequisite. For details see
issue with `libfuse` on Ubuntu >=22.04 or Debian 11 {libfuse2-url}[Setting up FUSE 2.x alongside of FUSE 3.x on recent Ubuntu (>=22.04), Debian and their derivatives].

> * The file browser integration, which means overlay icons and the context menu is not included in the AppImage. You need to install file browser extension packages manually.

Reviewer: ```suggestion
* The file browser integration, which means overlay icons and the context menu, is not included in the AppImage. You need to install file browser extension packages manually.
```

> * AppImages do not start automatically. You have to configure your desktop to automatically start the Desktop App App when logging in.

Reviewer: ```suggestion
* AppImages do not start automatically. You have to configure your desktop to automatically start the Desktop App when logging in.
```

** For GNOME, search for _startup applications_ in the desktop menu.

> ** As alternative, use the {appimagelauncher-url}[AppImageLauncher] App which also helps managing AppImages.

Reviewer: ```suggestion
** As an alternative, use the {appimagelauncher-url}[AppImageLauncher] App which also helps managing AppImages.
```

* There is no automatic updating. Any update is like installing the AppImage.

Installing _libfuse2_ if required::
--
* Check if `libfuse2` is already installed:
+
[source,bash]
----
dpkg -l libfuse2
----

* Check if there is an installation candidate for `libfuse2`:
+
[source,bash]
----
sudo apt-cache show libfuse2
----

* Install `libfuse2`:
+
[source,bash]
----
sudo apt install libfuse2
----
--

Install the AppImageLauncher app::

> See the {install-appimagelauncher-url}[Install AppImageLauncher] wiki for details installing it. The AppImageLauncher does not need to be started. It hooks in when you start and AppImage. There are different responses when starting an AppImage:

Reviewer: ```suggestion
See the {install-appimagelauncher-url}[Install AppImageLauncher] wiki for details about installing it. The AppImageLauncher does not need to be started. It hooks in when you start an AppImage. There are different responses when starting an AppImage:
```

+
--
How to open an AppImage::

> When you open an AppImage file via your file browser you have not opened before like double click on it:

Reviewer: I am not quite sure what this exactly is meant to say. Maybe?
```suggestion
When you open an AppImage file via your file browser that you have not opened before then double click on it:
```

+
image:installing/appimagelauncher_open_question.png[AppImage First Time Usage,width=350]
--
+
--
First time usage::
After opening an AppImage and the AppImage Launcher has been started for the first time, you will get asked to define some basic settings:
+
image:installing/appimagelauncher_first_run.png[AppImage First Time Usage,width=350]
--
+
--
AppImage Integration Question::

> Post first time configuration or when you open the AppImage file via your file browser like double clicking on it:

Reviewer: ```suggestion
Post first time configuration or when you open the AppImage file via your file browser, for example by double clicking on it:
```

+
image:installing/appimagelauncher_integrate_question.png[AppImage Integration Question,width=350]
--

Install and run the Desktop App AppImage::
The example below uses the terminal but you can also use the GUI. For details see {install-appimage-url}[How to run an AppImage].
+
--

> * Go to the {desktop-clients-url}[download] page and download the recent AppImage to your home directory into the Applications folder. Replace the URL from the example with the actual URL from the download page.

Reviewer: ```suggestion
* Go to the {desktop-clients-url}[download] page and download the recent AppImage into the Applications folder in your home directory. Replace the URL from the example with the actual URL from the download page.
```

> * Go to the {desktop-clients-url}[download] page and download the recent AppImage into the Applications folder in your home directory. Replace the URL from the example with the actual URL from the download page.

Author: ```suggestion
* Go to the {desktop-clients-url}[download] page and download the recent AppImage into the Applications folder in your home directory. Replace the URL from the example with the actual URL from the download page. Note the folder name `Applications` can be any name and helps to collect all AppImages you have on one location. The AppImageLauncher, if used, has this name predefined unless you change it.
```

+
[source,bash]
----
mkdir -p ~/Applications
----
+
[source,bash]
----
cd ~/Applications
----
+
[source,bash]
----
wget https://download.owncloud.com/desktop/ownCloud/stable//2.10.1.7187/linux-appimage/ownCloud-2.10.1.7187.AppImage
----

* The following steps are only necessary when the AppImageLauncher is _not_ used:
** Make the AppImage executable:
+
[source,bash]
----
sudo chmod +x ownCloud-2.10.1.7187.AppImage
----

* Start the AppImage by invoking the following command:
+
[source,bash]
----
~/Applications/ownCloud-2.10.1.7187.AppImage
----

> * Note when you start the AppImage post setting it to be executable, the AppImageLauncher will open if installed.

Reviewer: ```suggestion
* Note when you start the AppImage after setting it to be executable, the AppImageLauncher will open if installed.
```

Maybe?

--

== Customizing the Windows Installation

> If you just want to install ownCloud Desktop App on your local system, you can simply launch the `.msi` file and configure it in the wizard that pops up.

Reviewer: ```suggestion
If you just want to install the ownCloud Desktop App on your local system, you can simply launch the `.msi` file and configure it in the wizard that pops up.
```
